src/scripts/importFirebaseData.ts

```diff
@@ -0,0 +1,117 @@
+      photoUrl: string;
+      events: [
+        {
+          id: number;
+          isQuoteByNotablePerson?: true;
+          quote: string;
+          postedAt: number;
+          happenedOn?: number;
+          sourceName: string;
+          sourceUrl: string;
+          userComment: string;
+          userId: string;
+        }
+      ];
+    };
+  };
+  slugToID: {
+    [slug: string]: string;
+  };
+};
+
+connection
+  .then(async db =>
+    db.transaction(async entityManager => {
+      const users = db.getRepository(User);
+      let user = await users.findOne({ email: 'editor@hollowverse.com' });
+
+      if (!user) {
+        user = new User();
+        user.id = uuid();
+        user.fbId = '116989929051706';
+        user.email = 'editor@hollowverse.com';
+        user.name = 'Hollowverse Editor';
+        user.signedUpAt = new Date();
+        await entityManager.save(user);
+      }
+
+      const json = await readJson<FirebaseExport>('firebaseExport.json');
+
+      return Promise.all(
+        Object.entries(
+          json.notablePersons,
+        ).map(async ([id, { name, labels, events }]) => {
+          const notablePerson = new NotablePerson();
+          notablePerson.id = uuid();
+          notablePerson.name = name;
+          notablePerson.slug = findKey(json.slugToID, v => v === id)!;
+          notablePerson.photoId = notablePerson.slug;
+
+          notablePerson.labels = await entityManager.save(
+            labels.map(text => {
+              const label = new NotablePersonLabel();
+              label.id = uuid();
+              label.text = text;
+              label.createdAt = new Date();
+
+              return label;
+            }),
+          );
+
+          await entityManager.save(notablePerson);
+
+          await entityManager.save(
+            events.map(ev => {
+              const event = new NotablePersonEvent();
+              event.id = uuid();
+              event.sourceUrl = ev.sourceUrl;
+              event.isQuoteByNotablePerson = ev.isQuoteByNotablePerson || false;
+              event.quote = ev.quote;
+              event.happenedOn = ev.happenedOn ? new Date(ev.happenedOn) : null;
+              event.owner = user!;
+              event.postedAt = new Date(ev.postedAt);
+              event.notablePerson = notablePerson;
+
+              const comment = new NotablePersonEventComment();
+              comment.id = uuid();
+              comment.event = event;
+              comment.text = ev.userComment;
+              comment.owner = user!;
+              comment.postedAt = new Date(ev.postedAt);
+
+              event.comments = [comment];
+
+              return event;
+            }),
+          );
+
+          return notablePerson;
+        }),
+      );
+    }),
+  )
+  .then(() => {
+    console.info('Firebase data imported successfully');
+    process.exit(0);
+  })
+  .catch(e => {
+    console.error('Error importing data:', e.message || e);
+    process.exit(1);
+  });
```

Comment:
Do we still need this file?

Reply:
The data has only been imported into the development database attached to EB environment. The production database is still empty.